File: javascript/268. Missing Number/missingNumber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    var n = nums.length;
    var sum = n*(n+1)/2;

    nums.forEach((num) => {
        sum -= num;
    });

    return sum;
};


var nums = [0,1,2,3,5,6];

console.log(missingNumber(nums));

File: python/268. Missing Number/missing_number.py
def missing_number(nums):
    """
    :type nums: List[int]
    :rtype: int
    """

    n = len(nums)
    sum = n*(n+1)//2

    for num in nums: sum -= num

    return sum


if __name__ == "__main__":
    nums = [0,1,2,3,5,6]

    print(missing_number(nums))